cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -454,6 +483,12 @@ public Cancellable announce(String serviceName, int port) {
 
           // For binding StreamWriter
           install(createStreamFactoryModule());
+
+          // bind AuthorizationEnforcer to AuthorizationEnforcementService
+          bind(AuthorizationEnforcer.class).to(AuthorizationEnforcementService.class).in(Scopes.SINGLETON);
+          // also bind AuthorizationEnforcementService as a singleton. This binding is used while starting/stopping
+          // the service itself.
+          bind(AuthorizationEnforcementService.class).in(Scopes.SINGLETON);
```

Comment:
Seems like it only implements `AuthorizationEnforcer`. In that case, it's better to have two interfaces, `AuthorizationEnforcer` and `AuthorizationEnforcerService`, with the latter one extends from the first one. Then you have an implementation class `DefaultAithroizationEnforcerService` that implements the latter one. You can then have bindings like this:

```java
bind(AuthorizationEnforcer.class).to(AuthorizationEnforcerService.class);
bind(AuthorizationEnforcerService.class).to(DefaultAuthorizationEnforcerService.class).in(Scopes.SINGLETON);
```